pandas/core/aggregation.py

```diff
@@ -5,14 +5,46 @@
 
 from collections import defaultdict
 from functools import partial
-from typing import Any, DefaultDict, List, Sequence, Tuple
+from typing import Any, DefaultDict, List, Optional, Sequence, Tuple
 
 from pandas.core.dtypes.common import is_dict_like, is_list_like
 
+from pandas.core.base import SpecificationError
 import pandas.core.common as com
 from pandas.core.indexes.api import Index
 
 
+def reconstruct_func(func, *args, **kwargs):
+    """
+    This is the internal function to reconstruct func given if there is relabeling
```

Comment:
yeah pls format the doc-string, giving examples if possible.

Reply:
added docstring and examples